lisa_commands/commands.py

```diff
@@ -31,101 +29,169 @@ def parse_mention(message_text):
+            "description": "reverses text after command"
+        },
+        'shame': {
+            'function': shame,
+            "description": 'shame the text after command'
+        },
+        'sticker me': {
+            'function': sticker_me,
+            "description": "uses text after command to query giphey for stickers"
+        },
+        'table flip': {
+            'function': table_flip,
+            "description": "when you gotta flip a table"
+        },
+        'wiki me': {
+            'function': wiki_me,
+            "description": "use text after command to query wikipedia"
+        },
+        'youtube me': {
+            'function': youtube_me,
+            "description": "use text after command to query youtube"
+        },
+    }
+
+def run_command(command, query, slack_event):
+    """
+    runs the passed in command and query
+    :message_text contains text (sans bot mention)
+    :slack_event the event from slack
+    :Returns the result of the bot command
     """
+    command = command.lower().strip()
+    if command in commands().keys():
+        text = commands()[command]['function'](query, slack_event)
+        if text is None:
+            text = 'Sorry! No results found!'
+        return text
 
-    # for commands using the '* me' syntax (e.g. image me, youtube me, gif me etc.)
-    split = message_text.split(" me ", 1)
-    if len(split) == 2: return split
+    if command and command.strip().count(' ') > 0:
+        command, query_word = command.rsplit(' ', 1)
+        return run_command(command, query_word + ' ' + query, slack_event)
 
-    # If command isn't a '* me' command assume a single word command (e.g. decide, flipcoin, etc)
-    split = message_text.split(' ', 1)
-    if len(split) == 1: return [split[0].strip(), '']
+    return "Sorry I don't know that command. To see all my commands use `help`"
 
-    # if command is a single word command with text (e.g. decide, shame, etc)
-    if len(split) == 2: return split
+def anime_me(query, _slack_event):
+    """
+    query: query str
+    slack_event: A dict of slack event information(unused for this function)
+    returns the text to send back to slack
+    """
+    return request.anime_news_network_search('anime', query)
 
-    # else no command
-    return ['', '']
+def call_the_cops(_query, slack_event):
+    """
+    query: query str (unused for this function)
+    slack_event: A dict of slack event information
+    returns the text to send back to slack
 
+    """
+    return 'You called? ' + image_me('anime cops from bing', slack_event)
 
-def run_command(command, query, message_user):
+def decide(query, _slack_event):
     """
-    runs the passed in command and query
-    :command what to run (image me, gif me etc)
-    :query what to search for
-    :message_user the user who sent the request
-    :Returns the result of the bot command
+    query: query str
+    slack_event: A dict of slack event information(unused for this function)
```

Comment:
A little rusty on my python, is there a reason you need to pass in these unused `slack_event`'s to so many of these functions?

Reply:
Right the reason is because of this line here: https://github.com/Splyth/slack_bot/pull/10/files#diff-621cf83a34dcff0a31274dcb7b9de866R114

Here's kinda quick overview of what's going down (I can add this to a comment to make it clearer)

string gets passed in e.g. `image me a cute muffin`
it then tries to find that whole line in the commands() function
if it doesn't it chops off the last word and tries again. (e.g. image me a cute, then image me a, then finall image me which exists in the commands() dictionary)

Now here's the tricky part.
This is a recursive function and it dynamically calls the function from the `function` value for that command.

But I don't have a good way of knowing what function I'm calling. Some need a query some don't, some need slack_event info, some don't. So I gave everyone everything.

If you can think of a better way let me know

Reply:
I think it's fine, it doesn't affect performance, an it doesn't bog down the code.

Reply:
Another, less ideal option, would be to just grab the first word of the query, since they're all unique, and call the appropriate function based on that. Wouldn't scale very well though. I think what you have is fine.